src/tests/unit/doughs-dao.test.js

```diff
@@ -0,0 +1,85 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import sinon from 'sinon';
+
+import { getDoughs } from 'api/v1/db/oracledb/doughs-dao';
+import * as doughsSerializer from 'api/v1/serializers/doughs-serializer';
+import * as connectionApi from 'api/v1/db/oracledb/connection';
+// import { connection } from 'oracledb';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const baseGetDoughsReturn = {
+  data: [],
+};
+
+describe('test doughs dao', () => {
+  let serializeDoughsStub = null;
+  let getConnectionStub = null;
+  let connectionSpy = null;
+  const getDoughsConditions = 'ID AS "id", NAME AS "name", GRAMS_FLOUR AS "gramsFlour", GRAMS_WATER AS "gramsWater", FLOUR_TYPE AS "flourType", WATER_TEMP AS "waterTemp", GRAMS_YEAST AS "gramsYeast", GRAMS_SALT AS "gramsSalt", GRAMS_SUGAR AS "gramsSugar", GRAMS_OLIVE_OIL AS "gramsOliveOil", BULK_FERMENT_TIME AS "bulkFermentTime", PROOF_TIME AS "proofTime", SPECIAL_INSTRUCTIONS AS "specialInstructions"';
+  const getDoughsQuery = `SELECT ${getDoughsConditions} FROM DOUGHS `;
+  const getDoughsQueryWithWaterTemp = `SELECT ${getDoughsConditions} FROM DOUGHS WHERE WATER_TEMP = :waterTemp ;`;
+  const getDoughsQueryWithMultipleConditions = `SELECT ${getDoughsConditions} FROM DOUGHS WHERE WATER_TEMP = :waterTemp AND GRAMS_WATER = :gramsWater AND PROOF_TIME = :proofTime AND NAME = :name ;`;
+  const executeReturnRows = ['a', 'b', 'c'];
+  const executeReturn = { rows: executeReturnRows, metadata: 'ignore-this' };
+  beforeEach(() => {
+    connectionSpy = sinon.stub().returns(executeReturn);
+    serializeDoughsStub = sinon.stub(doughsSerializer, 'serializeDoughs');
+    getConnectionStub = sinon.stub(connectionApi, 'getConnection');
+    getConnectionStub.resolves({
+      execute: connectionSpy,
+      close: () => { },
+      commit: () => { },
+    });
+    serializeDoughsStub.returns(baseGetDoughsReturn);
+  });
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe('getDoughs', () => {
+    it('returns the output from serializeDoughs', () => {
+      const result = getDoughs({});
+      return result.should.eventually.deep.equal(baseGetDoughsReturn);
+    });
+    it('extracts only the rows from the database return', async () => {
+      await getDoughs({});
+      serializeDoughsStub.getCall(0).calledWith(executeReturnRows).should.equal(true);
```

Comment:
What does `true` mean? You should use `deepEqual` to compare the actual result. Example: https://github.com/osu-mist/terms-api/blob/b9acb21bb42fa5a483af15f514ea9017f4366c9f/src/tests/unit/terms-dao.test.js#L72-L74

Reply:
In this case it's testing that `calledWith` returns the literal value `true`.  I think that I can use the `.be.true` matcher to check that more explicitly.

Reply:
Hmm, not sure if I understand what you mean, but are you trying to assert if `serializeDoughsStub` is `calledWith` `executeReturnRows`? If so, you don't even need `should.equal(true)` or `should.be.true` since `calledWith` is already an assertion: https://www.chaijs.com/plugins/sinon-chai/

Reply:
Oh very cool, I'll definitely use the `should.have.been` matcher instead

Reply:
Cool 👍
Same with other comments I left, you probably also wanna assert what's the expected result of it.

Reply:
I think using `should.have.been.calledWith(...)` is the same as `calledWith(...).should.be.true` or `calledWith(...).should.deepEqual(true)` based on reading those docs

Reply:
Oh, I think I know what's the confusing part here lol.
Yeah it should be totally fine to use either `should.have.been.calledWith(...)` or `calledWith(...).should.be.true` to assert the spy object was called correctly. What I mean for `deepEqual` is we also need to asset the return result from `getDoughs(getDoughsData.mixedValidFilters)` is deeply equal to something.

Reply:
Ah, my original thought was that this test was only to check the values the spy was called with, and that checking the return value was handled in other tests.

Reply:
Oh, did I miss anything? Which test case tested the return value?

Reply:
The test `It('returns the output from serializeDoughs')` checks that, since the return value comes directly from `serializeDoughs` without any other processing.

Reply:
In that test case, we cover the case if the input `getDoughs()` is `{}`, right?

Reply:
That's true, I am going to add a test to make sure it returns the output of SerializeDoughs even with different inputs.